exporters/common/src/main/java/io/opentelemetry/exporter/internal/grpc/GrpcExporterBuilder.java

```diff
@@ -113,8 +115,16 @@ public GrpcExporterBuilder<T> setSslContext(
     return this;
   }
 
-  public GrpcExporterBuilder<T> addHeader(String key, String value) {
-    headers.put(key, value);
+  public GrpcExporterBuilder<T> addStaticHeader(String key, String value) {
+    if (staticHeaders == null) {
+      staticHeaders = new HashMap<>();
+    }
+    staticHeaders.put(key, value);
+    return this;
+  }
+
+  public GrpcExporterBuilder<T> setHeadersSupplier(Supplier<Map<String, String>> headerSupplier) {
```

Comment:
What do you say to a rename to something like `setExtraHeadersSupplier` or `setAdditionalHeadersSupplier`. The goal of a rename would be to eliminate any possibility that a reader might interpret the returned map as exhaustive. 

I think it could also help to have a note somewhere about how conflicts are handled -- the static headers take precedence.

Reply:
Was thinking about this a bit more..

With respect to how conflicts are handled, currently the PR prioritizes the static / constant heads over those from the supplier. But that doesn't need to be the case since since we can include multiple headers with the same key. So the conflict behavior should just be to include both.

But this makes me think the contract of `Supplier<Map<String, String>>` should actually be `Supplier<Map<String, List<String>>`. Why impost the constraint that header suppliers can provide multiple values for a particular header name?

> I think it could also help to have a note somewhere about how conflicts are handled

Yes, I should definitely update the javadoc of all the header setters on the exporter builders.

Reply:
I've pushed a commit to change the user API to set `Supplier<Map<String, List<String>>`, and to merge conflicts by including all header values instead of dropping one. Let me know what you think.

Reply:
I think it's a great improvement, thanks!